packages/list-store/src/search-params.ts

```diff
@@ -55,11 +59,30 @@ export function getUrlWithSearchParams({
     });
   }
 
-  // Only add relevant values to the search params. Remove all keys with empty strings as values
-  Object.keys(searchParams).forEach(key =>
-    searchParams[key] === '' ? delete searchParams[key] : {}
-  );
-  const encodedSearchParams = new URLSearchParams(searchParams).toString();
+  // Only add relevant values to the search params.
+  Object.keys(searchParams).forEach(key => {
+    if (searchParams[key] === '' || searchParams[key].length === 0) {
```

Comment:
Not sure whether this is an option, but is there a way to make these checks a part of the transformations on lines 72-83, so as to make the code more compact/sit together?

Reply:
Yes, with the new code, this makes more sense.